src/define.js

```diff
@@ -33,7 +34,7 @@ function define(/* [id], [dependencies], factory */) {
 
     tAMD._pre(function(id_, dependencies_, factory_) {
         run(function() {
-            addDefinition(id_, map(function(d) { return requireSync(d, id_, 1); }, dependencies_), factory_);
+            addDefinition(id_, map(function(d) {return requireSync(d, id_, 1);}, dependencies_), factory_, dependencies.indexOf('exports'));
```

Comment:
`dependencies.indexOf('exports')` should be changed to `dependencies_.indexOf('exports')` (with the trailing underscore).  This is because a 'define' hook might modify the dependency list - you want to look at the version of the list that has been processed by hooks.